services/backend/app/crud/crud_tags.py

```diff
@@ -13,9 +15,47 @@ class CRUDTag(CRUDBase[Tag, TagCreate, TagUpdate]):
     def get_all(self, db: Session) -> List[Tag]:
         return db.query(Tag).all()
 
+    def get_by_name(self, db: Session, name: str) -> Tag:
+        return db.query(Tag).filter(Tag.name == name).first()
+
+    def get_by_id(self, db: Session, id: UUID) -> Tag:
+        return db.query(Tag).filter(Tag.id == id).first()
+
     # TODO rename
     def get_page(self, db: Session, *, skip, limit) -> List[Tag]:
         return db.query(Tag).offset(skip).limit(limit)
 
+    def create(self, db: Session, *, obj_in: TagCreate) -> Tag:
+        db_obj = Tag(name=obj_in.name)
+        db.add(db_obj)
+        db.commit()
+        db.refresh(db_obj)
+        return db_obj
+
+    def update(
+        self, db: Session, *, db_obj: Tag, obj_in: Union[TagUpdate, Dict[str, Any]]
+    ) -> Tag:
+        if isinstance(obj_in, dict):
+            update_data = obj_in
+        else:
+            update_data = obj_in.dict(exclude_unset=True)
+        if update_data.get("name", False):
+            db_obj.name = update_data["name"]
+        if update_data.get("recipes", False):
```

Comment:
I'm not sure if it's a good idea for the `recipes` property to be modifiable.

Reply:
So you would only leave adding tags to recpie not recpie to tags?

Reply:
I don't really see any use for the latter. Tags are supposed to be immutable except for admins, and the UI probably will not have any way of adding recipes to tags.

Reply:
Fair